tt_metal/tools/profiler/op_profiler.hpp

```diff
@@ -235,26 +249,28 @@ namespace op_profiler {
         return fmt::format("`TT_DNN_FALL_BACK_OP:{}\n{}`",j["op_code"], ser);
     }
 
+    template<typename OutputTensors, template<typename> typename HostOperationType>
     inline std::string op_meta_data_serialized_json(
             uint32_t opID,
-            const tt::tt_metal::operation::HostOperation& op,
+            const HostOperationType<OutputTensors>& op,
             const std::vector<Tensor>& input_tensors,
-            std::vector<Tensor>& output_tensors)
+            OutputTensors& output_tensors)
     {
         auto j = get_base_json(opID, op, input_tensors, output_tensors);
         j["op_type"] = magic_enum::enum_name(OpType::tt_dnn_cpu);
         std::string ser = j.dump(4);
         return fmt::format("`TT_DNN_HOST_OP:{}\n{}`",j["op_code"], ser);
     }
 
+    template<typename OutputTensors, template<typename> typename DeviceOperationType>
     inline std::string op_meta_data_serialized_json(
             uint32_t opID,
             uint32_t device_id,
-            const tt::tt_metal::operation::DeviceOperation& op,
+            const DeviceOperationType<OutputTensors>& op,
             const std::variant<std::shared_ptr<Program>, std::reference_wrapper<Program>>& program,
             const std::vector<Tensor>& input_tensors,
             const std::vector<std::optional<const tt::tt_metal::Tensor>>& optional_input_tensors,
-            std::vector<Tensor>& output_tensors)
+            OutputTensors& output_tensors)
```

Comment:
CI right now does not do a through check on the generated op reports. Did you have a chance to sanity check an op-report and its input output tensor columns.

Reply:
It looks good to me.